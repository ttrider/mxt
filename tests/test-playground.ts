import { codegenSetup } from "./utils";
import path from "path";
import fs, { readFileSync, writeFileSync } from "fs";

const {parseWithComments} = require('jest-docblock');

const code = `
/**
 * This is a sample
 *
 * @flow foo
 * @bar [a,b,c]
 */
 
 console.log('Hello World!');
`;
 
const parsed = parseWithComments(code);

// prints an object with two attributes: comments and pragmas.
console.log(parsed);

describe("playground", () => {

  test("ex01", () => {
    const results = processTemplate("ex01"); 
    expect(results).toMatchSnapshot();
  });

  test("ex02", () => {
    const results = processTemplate("ex02");
    expect(results).toMatchSnapshot();
  });
  test("ex03", () => {
    const results = processTemplate("ex03");
    expect(results).toMatchSnapshot();
  });

});

function processTemplate(name: string) {

  const dir = path.resolve(__dirname, "..", "playground", "src", "templates");
  const inputFile = path.resolve(dir, name + ".html");
  const outputFile = path.resolve(dir, name + ".ts");

  const buffer = readFileSync(inputFile);
  if (buffer) {
    const results = codegenSetup(buffer.toString());
    writeFileSync(outputFile, results);

    return results;
  }

}
