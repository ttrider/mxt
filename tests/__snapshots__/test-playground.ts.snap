// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`playground ex01 1`] = `
"import $r$ from \\"../mxt-rt\\";
const { ex01 } = $r$({
    components: { ex01: \\"p1\\" },
    parts: { p1: $pf$ => {
            return {
                template: \\"\\\\n    <div id=\\\\\\"tagid_1\\\\\\">Hello MXT!</div>\\\\n\\",
                attachTo: [{
                        id: \\"tagid_1\\",
                        attrs: $dc$ => {
                            const { color } = $dc$.$data;
                            return { style: \`color: \${color}\` };
                        }
                    }]
            };
        } }
});
export { ex01 };
export default ex01;
"
`;

exports[`playground ex02 1`] = `
"import $r$ from \\"../mxt-rt\\";
const { ex02 } = $r$({
    components: { ex02: \\"p2\\" },
    parts: { p2: $pf$ => {
            return {
                template: \\"\\\\n    <div id=\\\\\\"tagid_2\\\\\\">Hello MXT!</div>\\\\n\\",
                attachTo: [{
                        id: \\"tagid_2\\",
                        attrs: $dc$ => {
                            const { color } = $dc$.$data;
                            return { style: \`color: \${color}\` };
                        },
                        events: [{
                                name: \\"click\\",
                                flags: 0,
                                handler: ($ev$: Event, $dc$: any) => {
                                    const { colorClick } = $dc$.$data;
                                    colorClick.bind($dc$.$data)($ev$, $dc$.$data, $dc$);
                                }
                            }]
                    }]
            };
        } }
});
export { ex02 };
export default ex02;
"
`;

exports[`playground ex03 1`] = `
"import $r$ from \\"../mxt-rt\\";
const { ex03 } = $r$({
    components: { ex03: \\"p3\\" },
    parts: { p3: $pf$ => {
            return {
                template: \\"\\\\n\\\\n    <style>\\\\n        div {\\\\n            background: #f0f0f0;\\\\n            cursor: pointer;\\\\n            box-sizing: content-box;\\\\n        }\\\\n\\\\n        div:hover {\\\\n            background: #e0e0f0;\\\\n        }\\\\n    </style>\\\\n\\\\n    <div id=\\\\\\"tagid_3\\\\\\">Hello MXT!</div>\\\\n\\",
                attachTo: [{
                        id: \\"tagid_3\\",
                        attrs: $dc$ => {
                            const { color } = $dc$.$data;
                            return { style: \`color: \${color}\` };
                        },
                        events: [{
                                name: \\"click\\",
                                flags: 0,
                                handler: ($ev$: Event, $dc$: any) => {
                                    const { colorClick } = $dc$.$data;
                                    colorClick.bind($dc$.$data)($ev$, $dc$.$data, $dc$);
                                }
                            }]
                    }]
            };
        } }
});
export { ex03 };
export default ex03;
"
`;
